doc/azure-ephemeral-disk-setup.8.in

```diff
@@ -0,0 +1,364 @@
+mount Azure ephemeral local disks (NVMe or SCSI) with optional RAID
+aggregation
+.SH SYNOPSIS
+.IP
+.nf
+\f[CR]
+azure-ephemeral-disk-setup
+\f[R]
+.fi
+.SH DESCRIPTION
+.PP
+\f[B]azure-ephemeral-disk-setup\f[R] is a systemd-managed one-shot
+initialization service for Microsoft Azure virtual machines that
+prepares ephemeral (temporary) disks for use.
+It supports both NVMe and SCSI ephemeral disks and is capable of
+aggregating multiple NVMe devices into a RAID-0 array using
+\f[CB]mdadm\f[R].
+.PP
+The service is opt-in and should be configured through
+\f[CB]/etc/azure-ephemeral-disk-setup.conf\f[R].
+It runs early in the boot process and aims to be idempotent, safe, and
+compatible with existing tools such as \f[CB]cloud-init\f[R] and
+\f[CB]WALinuxAgent\f[R].
+.PP
+Logs are printed to stderr and are accessible via journalctl if using
+the systemd service:
+.IP
+.nf
+\f[CR]
+journalctl -u azure-ephemeral-disk-setup.service
+\f[R]
+.fi
+.SH CONFIGURATION
+.PP
+Configuration is loaded from:
+.IP \[bu] 2
+\f[CB]/etc/azure-ephemeral-disk-setup.conf\f[R] (preferred)
+.IP \[bu] 2
+Environment variables (fallback)
+.PP
+The following environment variables are supported:
+.IP \[bu] 2
+\f[B]AZURE_EPHEMERAL_DISK_SETUP_AGGREGATION\f[R] Supported values:
+\f[CB]mdadm\f[R] Default: \f[CB]mdadm\f[R] (only supported value; enables
+aggregation of local NVMe disks when multiple are found)
+.IP \[bu] 2
+\f[B]AZURE_EPHEMERAL_DISK_SETUP_FS_TYPE\f[R] Supported values:
+\f[CB]ext4\f[R], \f[CB]xfs\f[R] Default: \f[CB]ext4\f[R] (universally
+available; \f[CB]xfs\f[R] requires \f[CB]xfsprogs\f[R])
+.IP \[bu] 2
+\f[B]AZURE_EPHEMERAL_DISK_SETUP_MDADM_CHUNK\f[R] Supported values: Any
+valid mdadm chunk size Default: \f[CB]512K\f[R] (mdadm default)
+.IP \[bu] 2
+\f[B]AZURE_EPHEMERAL_DISK_SETUP_MDADM_NAME\f[R] Supported values: Any
+valid mdadm name Default: \f[CB]azure-ephemeral-md\f[R]
+.IP \[bu] 2
+\f[B]AZURE_EPHEMERAL_DISK_SETUP_MOUNT_POINT\f[R] Supported values: Any
+valid mount point path Default: \f[CB]/mnt\f[R] (matches cloud-init
+default mount point)
+.IP \[bu] 2
+\f[B]AZURE_EPHEMERAL_DISK_SETUP_SCSI_RESOURCE\f[R] Supported values:
+\f[CB]true\f[R], \f[CB]false\f[R] Default: \f[CB]false\f[R] (minimizes risk
+of conflict with cloud-init and WALinuxAgent)
+.SH OPERATION
+.PP
+The service performs the following steps:
+.SS 1. Load and Validate Configuration
+.IP \[bu] 2
+Configuration is read via filtered
+\f[CB]source /etc/azure-ephemeral-disk-setup.conf\f[R], allowing only
+safe environment-style assignments.
+.IP \[bu] 2
+If invalid values are detected, the service exits with error.
+.IP \[bu] 2
+Required tools (\f[CB]mkfs.${AZURE_EPHEMERAL_DISK_SETUP_FS_TYPE}\f[R],
+\f[CB]mdadm\f[R]) must be available.
+.SS 2. Check for Existing Mounts
+.PP
+If the mount point is already configured:
+.IP \[bu] 2
+with a matching \f[CB]/etc/fstab\f[R] line containing the comment
+\f[CB]comment=azure-ephemeral-disk-setup\f[R], the service exits with
```

Comment:
consider documenting that it will re-format/raid/.... freshly allocated disk if the VM gets a new disk (deallocated/start or migration)

Reply:
done!